src/main/kotlin/com/grepp/nbe1_3_team04/team/service/TeamServiceImpl.kt

```diff
@@ -0,0 +1,126 @@
+        publisher.publishEvent(TeamPublishedEvent(createdTeam.name, createdTeam.teamId!!))
+
+        teamMemberRepository.save(TeamMember.createCreator(createdTeam, member))
+
+        return TeamDefaultResponse(createdTeam)
+    }
+
+
+    @Transactional(readOnly = true)
+    override fun getTeamInfo(teamId: Long): TeamInfoResponse {
+        //팀 정보
+
+        val teamEntity: Team = findTeamByIdOrThrowException(teamId)
+
+        //팀 평가 ->  List
+        val evaluations = teamRateRepository.findEvaluationsByTeam(teamEntity)
+            .map{it.evaluation ?: ""}
+
+        val maleCount = teamRepository.countMaleByMemberId(teamId)
+        val femaleCount = teamRepository.countFemaleByMemberId(teamId)
+
+        return TeamInfoResponse(
+            teamEntity,
+            evaluations,
+            maleCount,
+            femaleCount
+        )
+    }
+
+    @Transactional
+    override fun updateTeamInfo(teamId: Long, dto: TeamDefaultServiceRequest, member: Member): TeamDefaultResponse {
+        val memberId : Long = member.memberId ?: throw
+                IllegalArgumentException("존재하지 않는 회원입니다.")
+        //변경할 팀 id로 검색
+        val teamEntity: Team = findTeamByIdOrThrowException(teamId)
+        //현재 유저 정보 검색
+        val teamMember: TeamMember = findTeamMemberByIdOrThrowException(teamId, memberId)
+        //권한 정보
+        checkAuthority(teamId, teamMember)
+
+        //entity에 수정된 값 적용
+        dto.name?.let{teamEntity.updateName(it)}
+        dto.description?.let{teamEntity.updateName(it)}
+        dto.location?.let{teamEntity.updateName(it)}
+
+        //바뀐 Team값 반환
+        return TeamDefaultResponse(teamEntity)
+    }
+
+    @Transactional
+    override fun deleteTeam(teamId: Long, member: Member): Long {
+        val memberId : Long = member.memberId ?: throw
+            IllegalArgumentException("존재하지 않는 회원입니다.")
+        //삭제할 팀 탐색
+        val teamEntity: Team = findTeamByIdOrThrowException(teamId)
+        //현재 유저 정보 검색
+        val teamMember: TeamMember = findTeamMemberByIdOrThrowException(teamId, memberId)
+        //권한 정보
+        checkAuthority(teamId, teamMember)
+
+        teamRepository.delete(teamEntity)
+        // 채팅방 삭제 이벤트 실행
+        publisher.publishEvent(TeamDeletedEvent(teamId))
+        return teamId
+    }
+
+
+    fun findTeamByIdOrThrowException(id: Long): Team {
+        val team: Team = teamRepository.findByTeamId(id) ?: throw
+                IllegalArgumentException("해당 팀이 존재하지 않습니다.")
+        return team
+    }
+
+    fun findTeamMemberByIdOrThrowException(teamId: Long, memberId: Long): TeamMember {
+        val teamMember: TeamMember = teamMemberRepository.findByTeamIdAndMemberId(teamId, memberId) ?: throw
+                IllegalArgumentException("존재하지 않는 팀원입니다.")
+        return teamMember
+    }
+
+    fun checkAuthority(teamId: Long, teamMember: TeamMember) {
+        require(
+            !(teamMember.team.teamId !== teamId || teamMember.role !== TeamMemberRole.CREATOR)
```

Comment:
여기도 참조 동등성 하신 이유가 있나요? 값 비교로 불충분한가요?

Reply:
이미 위에서 언급하셨네요
그러면 저는 부정조건문 풀어보시는걸 추천드리겠습니다!
```suggestion
            teamMember.team.teamId == teamId && teamMember.role == TeamMemberRole.CREATOR
```